src/main/java/com/notably/model/block/BlockModel.java

```diff
@@ -0,0 +1,61 @@
+package com.notably.model.block;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * API of the BlockModel component.
+ */
+public interface BlockModel {
+    /**
+     * Gets the title of a block.
+     */
+    Title getTitle();
+
+    /**
+     * Gets the body content of a block.
+     */
+    Body getBody();
+
+    /**
+     * Gets the parent block of a block, if possible.
+     * Returns an {@code Optional<Empty>} if the block is the root block.
+     */
+    Optional<Block> getParent();
+
+    /**
+     * Gets a list of children blocks of a block.
+     */
+    List<Block> getChildren();
+
+    /**
+     * Replaces all the children of the block with a new list of children
+     */
+    void setChildren(List<Block> newChildren);
+
+    /**
+     * Finds a child block of a block, with the given input.
+     */
+    Optional<Block> getChild(Title title);
+
+    /**
+     * Replaces a child block of a block, with a new child block.
+     */
+    void setChild(Block oldBlock, Block newBlock);
```

Comment:
Should this `Block oldBlock` parameter instead be `Title title`?
If we take a look at `java.util.List`'s [set](https://docs.oracle.com/javase/8/docs/api/java/util/List.html) method, we can see that the parameters are `(int index, T item)`. Should we follow something similar here?

Reply:
Hmm, makes sense. I'll make the change

Reply:
Awesome!